src/pudl_rmi/create_override_spreadsheets.py

```diff
@@ -0,0 +1,708 @@
+
+    The process of manually checking all of the FERC-EIA matches made by the machine
+    learning algorithm is tedius. This function makes it easier to speed through the
+    obviously good matches and pay more attention to those that are more questionable.
+
+    By default, a "best match" is comprised of a FERC-EIA match with a capacity percent
+    difference of less than 6%, a net generation percent difference of less than 6%,
+    and an installation year difference of less than 3 years.
+
+    """
+    best_cap = df.capacity_mw_pct_diff < cap_pct_diff
+    best_net_gen = df.net_generation_mwh_pct_diff < net_gen_pct_diff
+    best_inst_year = df.installation_year_diff < inst_year_diff
+
+    df.loc[:, "best_match"] = df.best_match.fillna("")
+    df.loc[best_cap, "best_match"] = "cap"
+    df.loc[best_net_gen, "best_match"] = df.best_match + "_net-gen"
+    df.loc[best_inst_year, "best_match"] = df.best_match + "_inst_year"
+    df.loc[:, "best_match"] = df.best_match.replace(r"^_", "", regex=True)
+
+
+def _prep_ferc_eia(ferc1_eia, pudl_out) -> pd.DataFrame:
+    """Prep FERC-EIA for use in override output sheet pre-utility subgroups."""
+    logger.debug("Prepping FERC-EIA table")
+    check_connections = ferc1_eia[RELEVANT_COLS_FERC_EIA].copy()
+
+    # Add columns for user input and percent diff. These lists are specificalled
+    # ordered so that they appear in the df as they appear in each list.
+    ordered_input_values_cols = [
+        "verified",
+        "used_match_record",
+        "signature_1",
+        "signature_2",
+        "notes",
+        "record_id_override_1",
+        "record_id_override_2",
+        "record_id_override_3",
+        "best_match",
+    ]
+    ordered_pct_diff_cols = [
+        "fuel_type_code_pudl_diff",
+        "net_generation_mwh_pct_diff",
+        "capacity_mw_pct_diff",
+        "capacity_factor_pct_diff",
+        "total_fuel_cost_pct_diff",
+        "total_mmbtu_pct_diff",
+        "fuel_cost_per_mmbtu_pct_diff",
+        "installation_year_diff",
+    ]
+
+    for row in range(0, len(ordered_input_values_cols)):
+        check_connections.insert(row, ordered_input_values_cols[row], np.nan)
+
+    pct_diff_col_loc = check_connections.columns.get_loc("fuel_type_code_pudl_eia") + 1
+    for col in ordered_pct_diff_cols:
+        check_connections.insert(pct_diff_col_loc, col, np.nan)
+        pct_diff_col_loc = pct_diff_col_loc + 3
+
+    # Fix some column names
+    check_connections.rename(
+        columns={
+            "utility_id_pudl_ferc1": "utility_id_pudl",
+            "plant_id_pudl_ferc1": "plant_id_pudl",
+            "plant_name_new": "plant_name_eia",
+        },
+        inplace=True,
+    )
+
+    # Add pct diff columns
+    for col in [
+        "net_generation_mwh",
+        "capacity_mw",
+        "capacity_factor",
+        "total_fuel_cost",
+        "total_mmbtu",
+        "fuel_cost_per_mmbtu",
+    ]:
+        _pct_diff(check_connections, col)
```

Comment:
these little functions (this one and `_is_best_match`) that edit the df but don't return it. this is not a big deal but i think it would feel more explicit if you returned the df.

Reply:
ok! will do